conda_build/metadata.py

```diff
@@ -282,6 +282,32 @@ def get_contents(meta_path):
     return contents
 
 
+def handle_config_version(ms, ver):
+    """
+    'ms' is an instance of MatchSpec, and 'ver' is the version from the
+    configuration, e.g. for ms.name == 'python', ver = 26 or None,
+    return a (sometimes new) MatchSpec object
+    """
+    if ms.strictness == 3:
+        return ms
+
+    if ms.strictness == 2:
+        if ms.spec.split()[1] == 'x.x':
+            if ver is None:
+                raise RuntimeError("'%s' requires external setting" % ms.spec)
+            # (no return here - proceeds below)
+        else: # regular version
+            return ms
+
+    if ver is None or (ms.strictness == 1 and ms.name == 'numpy'):
```

Comment:
What special case?  The new function `handle_config_version()` is full of special cases, which is why add separated it out, in order to write extensive tests for it: https://github.com/conda/conda-build/blob/master/tests/test_metadata.py#L34